agent/app/agent.go

```diff
@@ -204,13 +207,13 @@ func (agent *ecsAgent) newTaskEngine(containerChangeEventStream *eventstream.Eve
 	if !agent.cfg.Checkpoint {
 		log.Info("Checkpointing not enabled; a new container instance will be created each time the agent is run")
 		return engine.NewTaskEngine(agent.cfg, agent.dockerClient,
-			credentialsManager, containerChangeEventStream, imageManager, state), "", nil
+			credentialsManager, containerChangeEventStream, imageManager, state, agent.metadataManager), "", nil
```

Comment:
I've addressed this issue with you in person, but I will say what I have done here to let other reviewers and interested parties of this pull request. 

The reason I extracted the metadata manager directly from the agent is that that is the practice already done earlier in that line of code for the client and config, for which the metadata manager is only a wrapper anyways. Furthermore, the alternative is to pass these parameters down from the ecsAgent.newTaskEngine function but since they are already available in the ecsAgent, that approach is redundant.

I have recently added an enable/disabling config variable to address the case when the metadataManager is missing or empty. Right now, because of how the metadataManager is defined as a wrapper for the docker client and agent config, the metadataManager cannot be invalid unless this feature is disabled; otherwise, there is also logic to terminate the agent if container metadata service is enabled but the metadataManager is invalid.